src/dirac_cwl_proto/execution_hooks/core.py

```diff
@@ -182,9 +317,39 @@ def get_output_query(self, output_name: str, **kwargs: Any) -> Optional[Path]:
         """Delegate to data catalog interface."""
         return self.data_catalog.get_output_query(output_name, **kwargs)
 
-    def store_output(self, output_name: str, src_path: str, **kwargs: Any) -> None:
-        """Delegate to data catalog interface."""
-        self.data_catalog.store_output(output_name, src_path, **kwargs)
+    def store_output(
+        self,
+        output_name: str,
+        src_path: str | Path | Sequence[str | Path],
+        **kwargs: Any,
+    ) -> None:
+        """Delegate to the correct interface."""
+        if "lfns_output_overrides" not in kwargs:
+            kwargs["lfns_output_overrides"] = self.lfns_output_overrides
+        if isinstance(src_path, Sequence) and not isinstance(src_path, str):
+            sb = []
+            for path in src_path:
+                if self.get_output_type(output_name, path) == OutputType.Sandbox:
+                    sb.append(path)
+                else:
+                    self.data_catalog.store_output(output_name, path, **kwargs)
+            if len(sb) > 0:
+                self.sandbox_interface.store_output(outputs=sb)
+        elif self.get_output_type(output_name, src_path) == OutputType.Sandbox:
+            self.sandbox_interface.store_output(outputs=[src_path])
+        else:
+            self.data_catalog.store_output(output_name, src_path, **kwargs)
+
+    def get_output_type(
+        self, output_name: str, src_path: str | Path, **kwargs: Any
+    ) -> OutputType:
+        """Whether the output must be stored in a Sandbox or the Data Catalog."""
+        if output_name in self.lfns_output_overrides or self.get_output_query(
+            output_name, **kwargs
+        ):
+            return OutputType.Data_Catalog
+        else:
+            return OutputType.Sandbox
```

Comment:
I might not understand it correctly, but this logic seems very different from what we have in DIRAC.
In DIRAC, in the JDL, we specifically specify which outputs we expect in the Data Catalog (`OutputData` section) and in the Output Sandbox (`OutputSandbox` section). 
Any particular reason for not reproducing this, or something similar, here?